### ISSUE TITLE
Calico eBPF + Metallb (with wireguard) custom interface connection refused when wireguard is down

### ISSUE THREAD

**Issue by ErGallo23** [projectcalico/calico#8626]
Hello,

I'm setting up a k0s cluster with calico and a custom wireguard network and I'm running into a problem when I failover test the wireguard network with the Load Balancer services. I'm using MetalLB as a load balancer system, using the Layer 2 option.

I have configured calico with VXLAN and eBPF, and the network nodes is wg0.

I have the load balancer service exposed on the local network interface (local interface  with IPs 192.168.XXX.XXX). I use this setting because I connect two nodes using the local interface as I don't want the traffic to go out through the wireguard network.

The problem appears when I simulate the fail of the wireguard network, and the nodes turn to the "NotReady" state. The local address 192.168.XXX.XXX stops responding with "connection refused" errors. This should not happen since if the wireguard network goes down, these nodes should still connect between them.

If, on the other hand, I disable eBPF and enable kube-proxy again and do the same test, the load balancer connection works.

I have tried several tests with "bpfDataIfacePattern" and "bpfL3IfacePattern" values ​​without success.

Is it possible to do what I'm trying with MetaLB's eBPF and Layer 2? If not, how could I emulate the behavior that DOES work with kube-proxy?

The current Felix configuration is:

 ```
 bpfConnectTimeLoadBalancing: TCP
  bpfEnabled: true
  bpfExternalServiceMode: Tunnel
  bpfHostNetworkedNATWithoutCTLB: Enabled
  bpfKubeProxyIptablesCleanupEnabled: false
  bpfLogLevel: debug
  floatingIPs: Disabled
  logSeverityScreen: Info
  reportingInterval: 0s
```

Thanks

**Comment by caseydavenport** [comment]
@tomastigera sounds like potentially something to do with our eBPF proxy implementation not matching upstream kube-proxy's behavior?

**Comment by tomastigera** [comment]
What is the nodes' internal IP address as reported by k8s? Which iface has that IP?

> I use this setting because I connect two nodes using the local interface as I don't want the traffic to go out through the wireguard network.

I am not sure I follow this. I understand that you have some traffic that does not go via wg, but I am not sure how that is achieved and what are the roles of these two nodes. Or are those the only 2 nodes you have?

Some sort of a diagram would help. What does routing look like?

What kinds of service uses metalb to advertise its IP? Is it a nodeport (likely)? Does the failure of the wg network mean that the IPs of the nodes changes? We implement forwarding of nodeport traffic via a vxlan tunnel (see bellow) and for that we need to know the node's IP. If that changes, it could break the service.

> I have tried several tests with "bpfDataIfacePattern" and "bpfL3IfacePattern" values ​​without success.

I was originally going to suggest that `bpfL3IfacePattern` might be something to look at. It seems like `wg0` definitelly needs to be part of it. The other option imo does not need to be changed. 

If you were able to tcpdump a connection when it works and when it does not, that would help as well.

![Screenshot from 2024-03-22 14-20-52](https://github.com/projectcalico/calico/assets/49207409/eb7214b3-4d75-4830-8c44-4c76c8997bc2)

**Comment by ErGallo23** [comment]
Hello,


 Thanks for the quick answer.  Yesterday I tried starting from 0 after you answered me using the default options of tigera-operator (previously I used the manifest installation) and it worked.  If the wireguard network went down, I could still access local services advertised via Layer 2 with metallb.


 It should be said that when I did the first tests I used the VXLAN backend (by default it is BIRD), or MTU 1420 (by default 0), among other parameter changes that may have made it not work.


 I close the issue since the problem seems more like a bad configuration on my part and not a calico error.


 Many thanks for everything
